soroban-sdk/src/env.rs

```diff
@@ -528,14 +530,42 @@ impl Env {
     ///
     /// # fn main() {
     /// let env = Env::default();
-    /// env.register_contract_wasm(WASM);
+    /// env.register_contract_wasm(None, WASM);
```

Comment:
Could you please add a comment that this replaces the code in case if it already exists? I don't think it's obvious from the name (same for another variant of this function).

Reply:
I'm going to leave the comment out, because I want to write more tests that actually confirm that replacing works, but this works given current behavior expected as is, so I'd like to merge it now for this release, and circle back to improve the testing.

i.e. This restores the existing API and existing behavior, and provides some potential new behavior that the docs aren't committing to (replacing).

I'll follow up this PR with tests around replacing.